modules/apps/27-interchain-accounts/module.go

```diff
@@ -97,25 +98,31 @@ type AppModule struct {
 	AppModuleBasic
 	controllerKeeper *controllerkeeper.Keeper
 	hostKeeper       *hostkeeper.Keeper
+
+	// legacyHostSubspace is used solely for migration of x/params managed parameters
+	legacyHostSubspace exported.Subspace
 }
 
 // NewAppModule creates a new IBC interchain accounts module
-func NewAppModule(controllerKeeper *controllerkeeper.Keeper, hostKeeper *hostkeeper.Keeper) AppModule {
+func NewAppModule(controllerKeeper *controllerkeeper.Keeper, hostKeeper *hostkeeper.Keeper, hss exported.Subspace) AppModule {
```

Comment:
This seems like an unnecessary field to add to the API since it will be removed. On one hand, it's nice to ensure the app.go file is providing the subspace, but it'd be nice if there's a way we could get the migrations to work by default without the app.go file changing?

Reply:
Maybe we should leave the subspace in the host keeper, add a `GetSubspace` function (since it is private) and remove the subspace from the keeper api after we do the migration. I'd prefer to limit app.go changes (otherwise upgrading ibc-go becomes a little more annoying) especially if those changes would need to be done again in the next release. Seems better to delay and have the removal of the subspace happen in the release after?

Reply:
I've done this change